src/greynirseq/utils/preprocessing/filters.py

```diff
@@ -0,0 +1,997 @@
+    lens_eng = [len(m) for m in prog.findall(eng)]
+    avg_ice = sum(lens_ice) / len(lens_ice)
+    avg_eng = sum(lens_eng) / len(lens_eng)
+    return avg_ice > 1.8 and avg_eng > 1.8
+
+
+@register_filter
+def missing_letter(ex):
+    # TODO(haukurb): gather file ids from this filter
+    ice = ex["is"]
+    substrings = [
+        r" a ",
+        r" e ",
+        r" i ",
+        r" o ",
+        r" u ",
+        r" y ",
+        r"\bvi\b",
+        r"\bess\b",
+        r"\bme\b",
+        r"\bessum\b",
+    ]
+    found = re.search("(" + "|".join(substrings) + ")", ice)
+    return not found
+
+
+@register_filter
+def alphanumeric(ex):
+    ice, eng = ex["is"], ex["en"]
+    prog = RegexCache.compile_rx(r"[^\d\W]")
+    mice = prog.search(ice)
+    meng = prog.search(eng)
+    return mice and meng
+
+
+@register_filter
+def sentence_length_ratio(ex):
+    ice, eng = len(ex["is"]), len(ex["en"])
+    res1 = ice > 3 or not (ice < 3 * eng)  # ice > 3 implies ice < 3 * eng
+    res2 = eng > 3 or not (eng < 3 * ice)
+    return res1 and res2
+
+
+@register_filter
+def strict_sentence_length_ratio(ex):
+    ice, eng = len(ex["is"]), len(ex["en"])
+    #res1 = ice > 3 or not (ice < 2 * eng)  # ice > 3 implies ice < 3 * eng
+    #res2 = eng > 3 or not (eng < 2 * ice)
+    res = (ice <= eng * 1.5) and (eng <= ice * 1.5)
+    return res
+
+
+@register_filter
+def token_count_ratio(ex):
+    if not T2T_AVAILABLE:
+        print(
+            "Skipping {0} because t2t is not available".format(
+                token_count_ratio.__name__
+            ),
+            file=sys.stderr,
+        )
+        return ex
+    enc = get_or_initialize_encoder()
+    ice = len(enc.encode(ex["is"]))
+    eng = len(enc.encode(ex["en"]))
+    # ice > n implies ice < k * eng is equivalent to
+    MIN_COUNT = 4
+    if ice < MIN_COUNT or eng < MIN_COUNT:
+        return True
+
+    res = (0.5 * ice < eng) and (0.5 * eng < ice)
+    return res
+
+
+@register_filter
+def case_mismatch(ex):
+    ice, eng = ex["is"], ex["en"]
+    ice = ice.upper() == ice
+    eng = eng.upper() == eng
+    return ice == eng
+
+
+@register_filter
+def digit_mismatch(ex):
+    prog = RegexCache.compile_rx(r"\D+")
+    # Remove non-digit characters, group consecutive numbers, filter empty string
+    ice = [w for w in prog.sub(" ", ex["is"]).strip(" ").split(" ") if w]
+    eng = [w for w in prog.sub(" ", ex["en"]).strip(" ").split(" ") if w]
+    dice = collections.Counter(ice)
+    deng = collections.Counter(eng)
+    mismatch = bool(dice - deng) or bool(deng - dice)
+    return not mismatch
+
+
+@register_filter
+def quote_mismatch(ex):
+    ice, eng = ex["is"], ex["en"]
+    prog = RegexCache.compile_rx(re.escape('"'))
+    nice = len(prog.findall(ice))
+    neng = len(prog.findall(eng))
+    return nice % 2 == 0 or neng % 2 == 0
+
+
+@register_filter
+def abs_min_string_edit(ex):
+    ice, eng = ex["is"], ex["en"]
+    dist = editdistance.eval(ice, eng)
+    return dist >= 3
```

Comment:
There is already an assumption elsewhere that no sequences are fewer than 4 words/tokens. But yes, this would reject any such examples.

With regards to magic constants, most or all of them belong in a config file that can be user-provided. Which was where this project is/was headed.